tests/integration/any_charm.py

```diff
@@ -0,0 +1,41 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""This code snippet is used to be loaded into any-charm which is used for integration tests."""
+import os
+import pathlib
+import signal
+import subprocess
+
+from any_charm_base import AnyCharmBase
+from ingress import IngressRequires
+
+
+class AnyCharm(AnyCharmBase):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.ingress = IngressRequires(
+            self,
+            {"service-hostname": self.app.name, "service-name": self.app.name, "service-port": 80},
+        )
+
+    def update_ingress(self, ingress_config):
+        self.ingress.update_config(ingress_config)
+
+    @staticmethod
+    def start_server(port: int = 80):
+        """Start an HTTP server daemon."""
+        www_dir = pathlib.Path("/tmp/www")
+        www_dir.mkdir(exist_ok=True)
+        ok_file = www_dir / "ok"
+        ok_file.write_text("ok")
+        pid_file = pathlib.Path("/tmp/any.pid")
```

Comment:
I'm guessing the intent for this is kind of a lock file to ensure only one http process is running at a time? Would a comment be beneficial here?

Reply:
Done

Reply:
This is not to prevent concurrent execution. The pid file exists because the Python HTTP server runs in daemon/separate mode, and without a pid file to record the PID of the HTTP server process, there would be no way to terminate the HTTP server.

Reply:
Sorry I explained things wrong in the previous comment, this is to prevent concurrent execution, but this is not a lock file. This file contains the current running process PID. When this file exist, the current running process will be killed before the new process started.